### PR TITLE
Entitlements validation endpoint

### PR DESCRIPTION
**Pull request by amitch23**
## Description

Adds an endpoint that will validate whether the weight estimate exceeds the maximum entitlement. It will be replicating the getEntitlements func in the client, but on the server, and returning a 200 if the weight estimate is within the entitlement, and a 409 response if not. This endpoint is hit on the Review screen, and if a 409 is returned, a notification/alert will pop up indicating what is to be done (e.g. the user has to lower the weight estimate on the edit weight screen). Their changes will be saved, and they will receive an Alert (pictured below).

If the weight estimate is below the entitlement but the entitlement has changed, they will receive an alert stating that (pictured below).

The backoffice can use this endpoint to validate changes to these models as well, and a-team should be notified of this endpoint when merged.

If we decide to go a different route, e.g. tying the validation to the models in question and returning custom responses and messages, we can reuse the logic in this handler then. There is some urgency to short circuit this edge case now, and if we go with this other way, there will need to be a good amount of tests written, and it may break backoffice code and/or add work to their current sprint. This is a step in getting entitlement validation where we eventually want it to be.

Note: this PR also adds the ability to pass custom messages to custom errors to the client.

## Code Review Verification Steps

* [x] All tests pass.
* [x] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [x] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [x] Request review from a member of a different team.
* [x] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/158082965) for this change

## Screenshots
Entitlement and weight estimate invalid:
screen shot 2018-06-13 at 10 17 05 pm

Entitlement has been changed:
screen shot 2018-06-13 at 10 17 18 pm

Returning custom message for the client in response body:
screen shot 2018-06-11 at 4 57 26 pm

### CODE REVIEW

**Comment by tinyels** [comment]
The cases where we need this notification, the user is not changing their weight estimate.  We need to notify the user when changes to their orders or service member information impacts their incentive. This can happen in 2 ways. The user can make changes on the review screen or the user might go back in the wizard and make changes. Design wants the notifications to happen on the review screen, so the user can then make the necessary weight adjustment.

File: pkg/models/order.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func (o *Order) Validate(tx *pop.Connection) (*validate.Errors, error) {
 		&validators.StringIsPresent{Field: string(o.Status), Name: "Status"},
 		&StringIsNilOrNotBlank{Field: o.TAC, Name: "Transportation Accounting Code"},
 		&StringIsNilOrNotBlank{Field: o.DepartmentIndicator, Name: "Department Indicator"},
+		&CannotHaveSpouseGearWithoutDependents{Field1: o.SpouseHasProGear, Name1: "SpouseHasProGear", Field2: o.HasDependents, Name2: "HasDependents"},
```
Comment (stangah) [inline pkg/models/order.go]: Could you generalize this to be something like `CannotHaveField1WithoutField2`? This could be useful elsewhere

Comment (amitch23) [reply pkg/models/order.go]: yes, good idea.

File: pkg/models/validators.go [hunk 2]
```diff
@@ -124,6 +124,21 @@ func (v *OrdersTypeIsPresent) IsValid(errors *validate.Errors) {
 	}
 }
 
+// CannotHaveField1WithoutField2 validates that spouseprogear cannot be true if has dependents is false
```
Comment (amitch23) [inline pkg/models/validators.go]: TODO: update comments

File: pkg/handlers/entitlements.go [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+	// Return 404 if there's no PPM or Rank
+	if len(move.PersonallyProcuredMoves) < 1 || serviceMember.Rank == nil {
+		return entitlementop.NewValidateEntitlementNotFound()
+	}
+	// PPMs are in descending order - this is the last one created
+	weightEstimate := *move.PersonallyProcuredMoves[0].WeightEstimate
+
+	smEntitlement := getEntitlement(*serviceMember.Rank, orders.HasDependents, orders.SpouseHasProGear)
+	if int(weightEstimate) > smEntitlement {
+		return responseForConflictErrors(h.logger, fmt.Errorf("your estimated weight of %s lbs is above your weight entitlement of %s lbs. You will only be paid for the weight you move up to your weight entitlement", humanize.Comma(weightEstimate), humanize.Comma(int64(smEntitlement))))
```
Comment (stangah) [inline pkg/handlers/entitlements.go]: Capitalize `your` plz

File: pkg/models/validators.go [hunk 4]
```diff
@@ -124,6 +124,21 @@ func (v *OrdersTypeIsPresent) IsValid(errors *validate.Errors) {
 	}
 }
 
+// CannotBeTrueIfFalse validates that spouseprogear cannot be true if has dependents is false
```
Comment (stangah) [inline pkg/models/validators.go]: Update comment plz

File: src/scenes/Review/Summary.jsx [hunk 5]
```diff
@@ -97,6 +102,18 @@ export class Summary extends Component {
             Your changes have been saved.
           </Alert>
         )}
+        {get(this.props.reviewState.error, 'statusCode', false) === 409 && (
+          <Alert type="warning" heading="Warning">
+            {this.props.reviewState.error.response.body.message}.
+          </Alert>
+        )}
+        {this.props.reviewState.entitlementChange &&
```
Comment (stangah) [inline src/scenes/Review/Summary.jsx]: Our plan for now is to just let this message persist, right? Until we figure out a long term strategy for alerting?

Comment (amitch23) [reply src/scenes/Review/Summary.jsx]: yes, that is correct, except only 1 message persists, the one that validates the entitlement. The notfication about the entitlement changing disappears after being displayed once.

File: pkg/models/validators.go (hunk 4 above)
Comment (amitch23) [reply pkg/models/validators.go]: done
